Sources/TuistCore/Models/TargetAction.swift

```diff
@@ -13,20 +13,62 @@ public struct TargetAction: Equatable {
         case post
     }
 
+    /// How to execute the target action
+    ///
+    /// - file: Executes the tool, calling the script at the path. Tuist will look up the tool on the environment's PATH
+    /// - text: Executes the entered script. This should be a short command.
+    private enum Script: Equatable {
+        case file(_ tool: String?, _ path: AbsolutePath?, _ args: [String])
+        case text(String)
+    }
+
     /// Name of the build phase when the project gets generated
     public let name: String
 
-    /// Name of the tool to execute. Tuist will look up the tool on the environment's PATH
-    public let tool: String?
+    /// The script to execute in the action
+    private let script: Script
+
+    public var isEmbeddedScript: Bool {
+        switch self.script {
+        case .file:
+            return false
+        case .text:
+            return true
+        }
+    }
+
+    /// Name of the tool to execute.
+    public var tool: String? {
+        switch script {
+        case .file(let tool, _, _):
```

Comment:
hmm, I don't think is completely right conceptually - if you specify a tool, you are not using a `file`. I think it should be parsed to `text` case or a new case altogether (that might make the most sense actually)

Reply:
Maybe I have the incorrect mental model here, but doesn't the `tool` in Tuist's case resolve to text entered in the Xcode script text box? Or is it in the `Shell` text box pictured below? If it's the former then using `tool`, `path`, and `arguments` is mutually exclusive to embedding a script since the embedded script would have all of those things in it already.

<img width="600" alt="Screen Shot 2020-12-23 at 10 37 31 AM" src="https://user-images.githubusercontent.com/2585841/103027512-dd511480-450a-11eb-9b57-3275bfe41cd0.png">

Reply:
yeah, it's rendered to that box. I'd create a new case then because  using a `tool` is exclusive with specifying a `path`, too

Reply:
Oh! Got it, that makes more sense to me now. Thanks!

Reply:
Hmmm when I do this I'm getting failures in the [`test_hash_targetAction_callsMockHasherWithExpectedStrings` test](https://github.com/tuist/tuist/blob/fd87d0ba7fef51e6340d9bbd4e641cdc28d4eb95/Tests/TuistCacheTests/ContentHashing/TargetActionsContentHasherTests.swift#L58). Looks like we are creating a TargetAction with both a path and a tool, which shouldn't be possible. Am I understanding that correctly?

Reply:
Yeah, I don't think that should be possible, I'd change the test

Reply:
Cool, will do.